nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js

```diff
@@ -2261,9 +2319,32 @@
          * @argument {object} properties        The properties
          * @argument {map} descriptors          The property descriptors (property name -> property descriptor)
          * @argument {map} history
+         * @argument {object} options
          */
-        loadProperties: function (properties, descriptors, history) {
-            return this.each(function () {
+        loadProperties: function (properties, descriptors, history, options) {
+            var self = this;
+            var groupId = null;
+
+            var loadParameterContext = function (options) {
+                if (typeof options.getFullParameterContextDeferred === 'function') {
```

Comment:
Is there a particular reason for passing `getFullParameterContextDeferred` and `getParameterContext` as properties of the `options` argument? Could they be passed as specific arguments instead of properties of `options`? If there is some other reason, recommend at least providing additional comments for the `options` argument.

Reply:
@exceptionfactory I added additional comments for the `options` argument. The `options` would only be necessary when `loadProperties` is called from processor configurations because it requires access to a parameter context in order to reference parameters. `loadProperties` can be called from other parts of NiFi, such as the Controller Service or Reporting Task that don't require additional options. In order to know this was called from processor configs, we can check for the parameter context options.